File: backend/app/schemas/course.py
from pydantic import BaseModel, field_validator
from typing import Optional, Literal
from datetime import date, datetime


class CourseCreate(BaseModel):
    name: str
    course_type: Optional[str] = None
    teacher_id: Optional[int] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    description: Optional[str] = None

    @field_validator("name")
    @classmethod
    def validate_name(cls, v: str) -> str:
        if not v or not v.strip():
            raise ValueError("课程名称不能为空")
        return v.strip()


class CourseUpdate(BaseModel):
    name: Optional[str] = None
    course_type: Optional[str] = None
    teacher_id: Optional[int] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    description: Optional[str] = None
    status: Optional[Literal["active", "completed", "cancelled"]] = None


class CourseResponse(BaseModel):
    id: int
    name: str
    course_type: Optional[str] = None
    teacher_id: Optional[int] = None
    teacher_name: Optional[str] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None
    description: Optional[str] = None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class CourseListResponse(BaseModel):
    items: list[CourseResponse]
    total: int
    page: int
    page_size: int
    total_pages: int

File: frontend/src/types/course.ts
export interface Course {
  id: number;
  name: string;
  course_type?: string;
  teacher_id?: number;
  teacher_name?: string;
  start_date?: string;
  end_date?: string;
  description?: string;
  status: string;
  created_at: string;
}

export interface CourseCreate {
  name: string;
  course_type?: string;
  teacher_id?: number;
  start_date?: string;
  end_date?: string;
  description?: string;
}

export interface CourseUpdate extends Partial<CourseCreate> {
  status?: 'active' | 'completed' | 'cancelled';
}

export interface CourseListResponse {
  items: Course[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface CourseListParams {
  page?: number;
  page_size?: number;
  search?: string;
  status?: string;
}
